File: dev c++的代码/判断素数.c
#include <stdio.h>
#include <stdlib.h>
#include<math.h>
int main()
{
	while(1)
	{
	int n;
	int isprime;
	scanf("%d",&n);
	int a[n];
	for(int i=0;i<n;i++)
	{
		scanf("%d ",&a[i]);
	}
	for(int i=0;i<n;i++)
	{
	isprime=judgeprime(a[i]);
	if(isprime==1)
	printf("1");
	else
	printf("0");
	if(i<n)
	printf(" ");
	}
	printf("\n");
	return 0;
}
int judgeprime(int n)
{
	int isprime=1;
	if(n==1||(n%2==0&&n!=2))
	isprime=0;
	else for(int i=3;i<=sqrt(n);i+=2)
	{
		if(n%i==0)
		isprime=0;
	}
	return isprime;	
} 

File: java作业/hello world/src/判断素数.java
import java.util.Scanner;

public class 判断素数 {
	static boolean judge_prime(int n)
	{
		//先假定某个数是素数，然后判断该数能不能被1和自身之外的数整除
		boolean isprime=true;
		//首先判断是不是2以外的偶数
		if(n%2==0&&n!=2)
		{
			isprime=false;
		}
		else
		{
			//由于2是素数，只需从3开始论证某个数不是素数，又根据数学公式，只需论证到n的1/2次方
			for(int i=3;i<=Math.sqrt(n);i+=2)
			{
				if(n%i==0)
				{
					isprime=false;
					break;
				}
			}
		}
		//至此已论证完某个数不是素数的所有可能性
		return isprime;
	}
	public static void main(String[] args) {
		Scanner in=new Scanner(System.in);
		int n;
		while(true)
		{
			System.out.print("请输入将要给判断的数：");
			n=in.nextInt();
			if(judge_prime(n))
			{
				System.out.println("是素数");
			}
			else
			{
				System.out.println("不是素数");
			}
		}

	}

}
